cmake/external/mkldnn.cmake

```diff
@@ -18,10 +18,15 @@ ENDIF(NOT ${WITH_MKLDNN})
 
 INCLUDE(ExternalProject)
 
-SET(MKLDNN_PROJECT "extern_mkldnn")
-SET(MKLDNN_SOURCES_DIR ${THIRD_PARTY_PATH}/mkldnn)
-SET(MKLDNN_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/opt/paddle/third_party/mkldnn")
-SET(MKLDNN_INCLUDE_DIR "${MKLDNN_INSTALL_DIR}/include" CACHE PATH "mkldnn include directory." FORCE)
+SET(MKLDNN_PROJECT        "extern_mkldnn")
+SET(MKLDNN_SOURCES_DIR    ${THIRD_PARTY_PATH}/mkldnn)
+SET(MKLDNN_INSTALL_ROOT   ${CMAKE_INSTALL_PREFIX})
+IF(NOT "$ENV{HOME}" STREQUAL "/root")
+    SET(MKLDNN_INSTALL_ROOT  "$ENV{HOME}")
+ENDIF()
+
+SET(MKLDNN_INSTALL_DIR    "${MKLDNN_INSTALL_ROOT}/opt/paddle/third_party/mkldnn")
```

Comment:
Yes, I can get your point~
But as those *.so needed should be placed in the appropriate folder under permission.
And refer to gpu( `/usr/lib`) and mkl`(opt/intel/mkl`), their so files are all in special folders that would not be removed easily.